.github/workflows/Publish_NIMS.yml

```diff
@@ -54,15 +54,19 @@ jobs:
         working-directory: ./ni_measurementlink_generator
 
       - name: Commit file changes
+        id: commit
         if: ${{ startsWith(github.event.release.target_commitish, 'main') || startsWith(github.event.release.target_commitish, 'releases/') }}
         run: |
-          git config --local user.email "action@github.com"
-          git config --local user.name "GitHub Action"
-          git add .
-          git commit -m "Promote NIMS package version and update Sphinx generated files in _docs" -a
+          echo "version_changed=0" >> $GITHUB_OUTPUT
+          if [ -n "$(git status --porcelain)" ]; then
+            git config --local user.email "action@github.com"
+            git config --local user.name "GitHub Action"
+            git add .
+            git commit -m "Promote NIMS and NIMG package version" -a
+            echo "version_changed=1" >> $GITHUB_OUTPUT
       
       - name: Push changes to the appropriate branch
-        if: ${{ startsWith(github.event.release.target_commitish, 'main') || startsWith(github.event.release.target_commitish, 'releases/') }}
+        if: ${{ steps.commit.outputs.version_changed==1 && (startsWith(github.event.release.target_commitish, 'main') || startsWith(github.event.release.target_commitish, 'releases/')) }}
```

Comment:
Use a boolean so you don't have to compare to 1.

https://docs.github.com/en/actions/learn-github-actions/expressions